### DISCUSSION TITLE
Persistent cURL Error 7: docker container connection issues across services

### DISCUSSION THREAD

**Discussion by leonardodeglinnocenti** [nextcloud/all-in-one#4430] [Questions] [answered]
I have a very serious connection problem with many external services. I noticed this frequent error log:
`[fulltextsearch_elasticsearch] Error: Retry 0: cURL error 7: Failed to connect to nextcloud-aio-fulltextsearch port 9200 after 13 ms: Couldn't connect to server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://elastic:<redacted-token-here>@nextcloud-aio-fulltextsearch:9200/nextcloud-aio/_doc/files%3A1165878?pipeline=attachment
	from ? by -- at 23 mar 2024, 21:00:42
`
I found it was discussed here https://github.com/nextcloud/all-in-one/discussions/4290, but the answer marked as such is not actually helpful in any way, and I suppose this problem is not isolated given the other user's replies. This definitely requires a deeper analysis as this is an issue that generates a lot of frequent errors in the logs.
This is not the only connection problem that I have with a docker container, as a similar issue is found in "local-ai", the external Memories transcoder and the Talk recording server.
1. Talk recording backend:
`[no app in context] Error: Exception thrown: GuzzleHttp\Exception\ConnectException
	POST /ocs/v2.php/apps/spreed/api/v1/recording/3m8eo4ju
	from <ip> by leonardo at 23 mar 2024, 21:19:27
`
2. LocalAI (strangely it refers to v1):
`[integration_openai] Warning: API request error : cURL error 7: Failed to connect to nextcloud-aio-local-ai port 8080 after 13 ms: Couldn't connect to server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://nextcloud-aio-local-ai:8080/v1/chat/completions
	POST /ocs/v2.php/apps/assistant/api/v1/task/run-or-schedule
	from <ip> by leonardo at 23 mar 2024, 16:58:19
`
3. Memories transcoding community container:
`[memories] Error: Transcode failed: failed to connect to go-vod: cURL error 7: Failed to connect to nextcloud-aio-memories port 47788 after 12 ms: Couldn't connect to server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://nextcloud-aio-memories:47788/config%2Fconfig/config
	GET /apps/memories/api/video/transcode/w21pq7obff00/993084/index.m3u8
	from <ip> by leonardo at 23 mar 2024, 10:12:04
`

The **cURL error 7** is recurring very frequently. The connection with external docker containers seems broken. Collabora server works correctly, while I had to disable ClamAV because it generated a huge amount of logs (but maybe this problem is not correlated). In the Nextcloud AiO Interface 8.0.0 all containers are correctly running.

### My configuration details:
**Operating system:** Linux 5.10.0-28-amd64 SMP Debian 5.10.209-2 (2024-01-31) x86_64

**Webserver:** Apache/2.4.58 (Unix) (fpm-fcgi)

**Database:** pgsql PostgreSQL 15.6 on x86_64-pc-linux-musl, compiled by gcc (Alpine 13.2.1_git20231014) 13.2.1 20231014, 64-bit

**PHP version:** 8\.2.16

Modules loaded: Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, hash, iconv, json, mbstring, SPL, session, PDO, pdo_sqlite, bz2, posix, random, readline, Reflection, standard, SimpleXML, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, cgi-fcgi, apcu, bcmath, Phar, exif, ftp, gd, gmp, imagick, imap, intl, ldap, memcached, pcntl, pdo_pgsql, pgsql, redis, smbclient, sodium, sysvsem, zip, libsmbclient, Zend OPcache

**Nextcloud version:** 28\.0.3 - 28.0.3.2

**Encryption:** no

**User-backends:**

* OC\\User\\Database*

**Cron Configuration:** Array
(
[backgroundjobs_mode] => cron
[lastcron] => 1711230016
)

**External storages:** yes

**Talk configuration:**

STUN servers

* <url>:3478

TURN servers

* turn:<url>:3478 - udp,tcp

Signaling servers (mode: default):

* SIP dialin is disabled
* SIP dialout is disabled
* <url>/standalone-signaling/ - 1.2.3~docker

Recording servers:

* Recording is enabled
* Recording consent is set to "default"
* http://nextcloud-aio-talk-recording:1234/ - error: cURL error 7: Failed to connect to nextcloud-aio-talk-recording port 1234 after 12 ms: Couldn't connect to server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://nextcloud-aio-talk-recording:1234/api/v1/welcome

**Browser:** Mozilla/5.0 (X11; Linux x86_64; rv:124.0) Gecko/20100101 Firefox/124.0

**Comment by szaimen** [comment]
Hi, all of these issues sound like docker networking related which is not caused by AIO but by the underlying infrastructure like your docker installation and/or firewall or others.

Can you follow https://github.com/nextcloud/all-in-one/discussions/2557 and check if that improves things?

**Reply by szaimen** [reply]
Apparently this might be a bug: https://github.com/nextcloud/all-in-one/issues/4440

**Reply by leonardodeglinnocenti** [reply]
Unfortunately #2557 does not solve the problem. Even the steps described in #4440 to replace the curl version don't work.

**Reply by leonardodeglinnocenti** [reply]
I tried to completely reinstall LocalAI by removing the container and pruning the related volumes. Unfortunately it didn't solve the problem. Still the cURL error persists in all external services.

**Comment by szaimen** [comment, marked as answer]
Hi, can you upgrade to v9.1.0 beta and check if that fixes the problem for you? See https://github.com/nextcloud/all-in-one

**Reply by leonardodeglinnocenti** [reply]
Hi, I updated the system to 9.0.1 as soon as the update was available. As I use Nextcloud to study, I need high stability and I would prefer to wait for 9.1.0 to enter the stable branch. The `cURL Error 7` is gone, but unfortunately it has been replaced by:
`[fulltextsearch_elasticsearch] Error: Retry 0: cURL error 52: Empty reply from server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://elastic:765b3cf6aceaed18a4d90e06c71f2937c7565465d4fd85f3@nextcloud-aio-fulltextsearch:9200/nextcloud-aio/_doc/files%3A1187421?pipeline=attachment
	from ? by -- at 28 giu 2024, 14:12:55
`
which is coupled with
`{"reqId":"NAFA0XsFQKA4Pv1BopZg","level":2,"time":"2024-06-28T11:57:54+00:00","remoteAddr":"","user":"--","app":"fulltextsearch","method":"","url":"--","message":"platform seems down. we will update index next cron tick","userAgent":"--","version":"29.0.2.2","data":{"app":"fulltextsearch"},"id":"667eaa13bc78c"}`

All connectivity issues with external services seem solved (even if for some reason Memories transcoding is still broken on my installation). The reliability of Talk has worsened in case of client unstable connection (probably related to the HPB).
